### PR TITLE
boards: espressif: Add support for ESP32-C3 Super Mini board

### PR DESCRIPTION
**Pull request by dawr68**
This PR add support for ESP32-C3 Super Mini board.

Verified on real hardware by the test codes.

### CODE REVIEW

File: boards/espressif/esp32c3_supermini/board.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: esp32c3_supermini
+  vendor: espressif
```
Comment (marekmatej) [inline boards/espressif/esp32c3_supermini/board.yml:3]: "Espressif" doesn't seem like correct vendor.

File: boards/espressif/esp32c3_supermini/doc/img/esp32c3-supermini-pinout.jpg
Comment (nordicjm) [inline boards/espressif/esp32c3_supermini/doc/img/esp32c3-supermini-pinout.jpg:1]: do we really need a pinout here, can you link to it on the manufacturer's website?

File: boards/espressif/esp32c3_supermini/doc/img/esp32c3-supermini.jpg
Comment (nordicjm) [inline boards/espressif/esp32c3_supermini/doc/img/esp32c3-supermini.jpg:1]: Convert to webp and put through https://tinypng.com/ should be 50KiB at most

File: boards/espressif/esp32c3_supermini/doc/img/esp32c3-supermini-schematic.jpg
Comment (nordicjm) [inline boards/espressif/esp32c3_supermini/doc/img/esp32c3-supermini-schematic.jpg:1]: do we really need a schematic here, can you link to it on the manufacturer's website?

File: boards/espressif/esp32c3_supermini/esp32c3_supermini.dts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		slot1_partition: partition@110000 {
+			label = "image-1";
+			reg = <0x00110000 0x00100000>;
+		};
+
+		/* Reserve 256kB for the scratch partition */
+		scratch_partition: partition@210000 {
+			   label = "image-scratch";
+			   reg = <0x00210000 0x00040000>;
+		};
```
Comment (nordicjm) [inline boards/espressif/esp32c3_supermini/esp32c3_supermini.dts:137]: really shouldn't be using scratch partition in 2024

**Review by marekmatej — CHANGES_REQUESTED** [review]
I like the board because of its small form factor. But sadly, this board is not from the Espressif board portfolio. Therefore I suggest changing the vendor and board location folder. Also, there are other things that should be updated:
- Copyright messages
- Documentation does not fit the latest HALv5 changes (will be updated upstream)
- some commits need to be squashed

I suggest waiting until HALv5 will be upstreamed and then submit the work again.

**Pull request closed without merging**